### DISCUSSION TITLE
Entities do not move?

### DISCUSSION THREAD

**Discussion by gigabitwize** [Minestom/Minestom#1729] [Help]
Build: 1.19-6c1946c489-1

I've made a custom entity Zombie and gave it an AI group which includes targeting and attacking a player, but the entity doesn't move at all. After that I tried giving it a RandomStrollGoal, but that also did not make the entity move.

Ultimately I tried to move the entity manually using getNavigator#moveTowards after spawning it, but the entity still does not move.

Am I doing something wrong on my end?

```java
public Monster(Instance instance, Pos pos, String displayName, @NotNull EntityType entityType) {
        super(entityType);
        setCustomName("Monster"));
        setCustomNameVisible(true);

        getAttribute(Attribute.MAX_HEALTH).setBaseValue(100);
        heal();

        addAIGroup(new EntityAIGroupBuilder()
                .addGoalSelector(new RandomStrollGoal(this, 10))
                .addGoalSelector(new RandomLookAroundGoal(this, 50))
                .build());

        setInstance(instance, pos);
    }
```
This used to work fine in the past.

**Comment by KrystilizeNevaDies** [comment]
running `setInstance(instance, pos);` in the constructor is not a good idea, that may cause bugs. (It may be causing this one)

**Reply by gigabitwize** [reply]
Hey, thank you for the response. That doesn't seem to be the issue, using setInstance outside of the constructor doesn't fix it for me.

**Comment by gigabitwize** [comment]
I took a look at the Minestom arena code to see whether I'm doing anything wrong, and it doesn't seem to be the case.

I basically copied the following:
```java
        EntityCreature entityCreature = new EntityCreature(EntityType.ZOMBIE);
        entityCreature.addAIGroup(
                List.of(new MeleeAttackGoal(entityCreature, 1.2, 20, TimeUnit.SERVER_TICK)),
                List.of(new ClosestEntityTarget(entityCreature, 32, entity -> entity instanceof Player))
        );

        entityCreature.setInstance(instance, pos);
```
And the entity does not move at all. This seems to be a bug in the latest build?

Can anyone verify?

**Comment by gigabitwize** [comment]
For those that are in a similar situation, please check my last comment in my other issue: https://github.com/Minestom/Minestom/issues/1730

That should fix it.
